image/pict.cpp

```diff
@@ -344,13 +350,26 @@ void PICTDecoder::unpackBitsRect(Common::SeekableReadStream &stream, bool withPa
 
 	// Read in amount of data per row
 	for (uint16 i = 0; i < packBitsData.pixMap.bounds.height(); i++) {
-		// NOTE: Compression 0 is "default". The format in SCI games is packed when 0.
-		// In the future, we may need to have something to set the  "default" packing
+		// NOTE: Compression 0 is "default". The format in SCI games is packed when 0
+		// unless rowBytes is less than 8 in which case the pict can't be packed,
+		// such as the shovel inventory icon in FPFP Mac. (bug #7059)
+		// In the future, we may need to have something to set the "default" packing
 		// format, but this is good for now.
 
 		if (packBitsData.pixMap.packType == 1 || packBitsData.pixMap.rowBytes < 8) { // Unpacked, Pad-Byte (on 24-bit)
-			// TODO: Finish this. Hasn't been needed (yet).
-			error("Unpacked DirectBitsRect data (padded)");
+			// only support 1 bpp for now as there is currently only one known
+			//  SCI pict that requires any unpacked support.
+			if (bytesPerPixel == 1 && packBitsData.pixMap.pixelSize == 8) {
+				for (uint16 b = 0; b < packBitsData.pixMap.rowBytes; b++) {
+					byte pixel = stream.readByte();
+					if (b < width) {
+						buffer[i * width + b] = pixel;
+					}
+				}
```

Comment:
You don’t need the for loop here, since this is just a direct copy of data.

You could just read all the bytes like this:
stream.read(buffer[i * width], packBitsData.pixMap.rowBytes);

Not sure why the width check is needed in this case, but you could use MIN() for that, if you want to keep it

Reply:
Got rid of the loop, that's cleaner, but here's what's going on...

"width" and packBitsData.pixMap.rowBytes aren't always the same.

In the case of the shovel, there's a zero byte at the end of each row. This is padding that may or may not appear (alignment? 24-bit? see existing vague comment)

Even without the padding, these values might be different. The purpose of this opcode family is to copy bits and clip them to a rectangle. "width" has that clipping factored in, so they could be different for that reason too.

In the shovel's case, we need to consume 6 bytes from the stream but only apply 5 to buffer on each row.

This should cover all cases.